components/eam/src/physics/cam/microp_aero.F90

```diff
@@ -1,789 +1,686 @@
-
-   !! interface 
-
-   integer,  intent(in) :: ncol              ! number of cols 
-   real(r8), intent(in) :: wsig(pcols,pver ) ! standard deviation (m/s)
-   real(r8), intent(in) :: w0(pcols,pver ) ! large scale vertical velocity (m/s) 
-   real(r8), intent(out):: ww(pcols,pver) ! mean updraft velocity(m/s) -> characteristic w*
-
-   !! local 
-   integer, parameter :: nbin = 50
-
-   real(r8) :: wlarge,sigma
-   real(r8) :: xx, yy 
-   real(r8) :: zz(nbin) 
-   real(r8) :: wa(nbin) 
-   integer  :: kp(nbin) 
-   integer  :: i, k
-   integer  :: ibin
-
-   !! program begins 
-
-   do k = 1, pver
-   do i = 1, ncol
-
-      sigma  = max(0.001_r8, wsig(i,k))
-      wlarge = w0(i,k)
-
-      xx = 6._r8 * sigma / nbin
-
-      do ibin = 1, nbin
-         yy = wlarge - 3._r8*sigma + 0.5*xx
-         yy = yy + (ibin-1)*xx
-         !! wbar = integrator < w * f(w) * dw > 
-         zz(ibin) = yy * exp(-1.*(yy-wlarge)**2/(2*sigma**2))/(sigma*sqrt(2*pi))*xx
-      end do 
-
-      kp(:) = 0 
-      wa(:) = 0._r8 
- 
-      where(zz.gt.0._r8) 
-         kp = 1 
-         wa = zz
-      elsewhere 
-         kp = 0 
-         wa = 0._r8 
-      end where 
-
-      if(sum(kp).gt.0) then 
-         !! wbar = integrator < w * f(w) * dw > 
-         ww(i,k) = sum(wa)
-      else 
-         ww(i,k) = 0.001_r8
-      end if 
-
-      !!write(6,*) 'i, k, w0, wsig, ww : ', i, k, w0(i,k), wsig(i,k), ww(i,k) 
-
-  end do
+  !---------------------------------------------------------------------------------
+  ! Purpose: Calculate the mean updraft velocity inside a GCM grid assuming the 
+  !          vertical velocity distribution is Gaussian and peaks at the 
+  !          GCM resolved large-scale vertical velocity. 
+  !          When icenul_wsub_scheme = 2, the model uses the mean updraft velocity as the 
+  !          characteristic updraft velocity to calculate the ice nucleation rate. 
+  ! Author:  Kai Zhang (kai.zhang@pnnl.gov) 
+  ! Last Modified: Oct, 2015 
+  !---------------------------------------------------------------------------------
+
+  !! interface 
+
+  integer,  intent(in) :: ncol              ! number of cols 
+  real(r8), intent(in) :: wsig(pcols,pver ) ! standard deviation (m/s)
+  real(r8), intent(in) :: w0(pcols,pver ) ! large scale vertical velocity (m/s) 
+  real(r8), intent(out):: ww(pcols,pver) ! mean updraft velocity(m/s) -> characteristic w*
+
+  !! local 
+  integer, parameter :: nbin = 50
+
+  real(r8) :: wlarge,sigma
+  real(r8) :: xx, yy 
+  real(r8) :: zz(nbin) 
+  real(r8) :: wa(nbin) 
+  integer  :: kp(nbin) 
+  integer  :: i, k
+  integer  :: ibin
+
+  !! program begins 
+
+  do k = 1, pver
+     do i = 1, ncol
+
+        sigma  = max(0.001_r8, wsig(i,k))
+        wlarge = w0(i,k)
+
+        xx = 6._r8 * sigma / nbin
+
+        do ibin = 1, nbin
+           yy = wlarge - 3._r8*sigma + 0.5*xx
+           yy = yy + (ibin-1)*xx
+           !! wbar = integrator < w * f(w) * dw > 
```

Comment:
this line (and a few others below) may be removed.

Reply:
This line seems like a comment about the code operation, so I am keeping it for now.